packages/hash/frontend/src/blocks/onBlockLoaded.tsx

```diff
@@ -0,0 +1,80 @@
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { getBlockDomId } from "./page/BlockView";
+
+type OnBlockLoadedFunction = (blockEntityId: string) => void;
+
+/** @private enforces use of custom provider */
+const BlockLoadedContext = createContext<OnBlockLoadedFunction | null>(null);
+
+export const BlockLoadedProvider: React.FC<{ routeHash: string }> = ({
+  routeHash,
+  children,
+}) => {
+  const [hashToScrollTo, setHashToScrollTo] = useState<string | null>(null);
```

Comment:
I don't believe you need this variable – its seemingly just used to trigger an effect which can just happen in the `onBlockLoaded` callback instead